File: lib/data/models/player.dart
class Player {
  final String id;
  final String name;
  final int age;
  final String photoUrl;
  final int wins;
  final int losses;
  final int matchesPlayed;

  Player({
    required this.id,
    required this.name,
    required this.age,
    required this.photoUrl,
    this.wins = 0,
    this.losses = 0,
    this.matchesPlayed = 0,
  });

  factory Player.fromJson(Map<String, dynamic> json) {
    return Player(
      id: json['_id']?.toString() ?? '', // Gestion du null
      name: json['name']?.toString() ?? 'Unknown', // Conversion sécurisée
      age: (json['age'] as int?) ?? 0,
      photoUrl:
          json['photoUrl']?.toString() ?? 'https://via.placeholder.com/150',
      wins: (json['wins'] as int?) ?? 0,
      losses: (json['losses'] as int?) ?? 0,
      matchesPlayed: (json['matchesPlayed'] as int?) ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {'name': name, 'age': age, 'photoUrl': photoUrl};
  }
}

File: src/models/Player.js
const mongoose = require('mongoose');

const playerSchema = new mongoose.Schema({
  name: { 
    type: String, 
    required: true,
    trim: true
  },
  age: { 
    type: Number, 
    required: true,
    min: 10,
    max: 60
  },
  photoUrl: { 
    type: String,
    default: 'https://via.placeholder.com/150'
  },
  wins: {
    type: Number,
    default: 0
  },
  losses: {
    type: Number,
    default: 0
  },
  matchesPlayed: {
    type: Number,
    default: 0
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
});

module.exports = mongoose.model('Player', playerSchema);